### ISSUE TITLE
When node in memory pressure,static pod enters Error status after node reboot

### ISSUE THREAD

**Issue by Decade2014** [kubernetes/kubernetes#109433]
### What happened?

Conditions:
1. node is in status of memory pressure;
2. node reboot.

What happened?
1. static pod enters Error status and never restore Running status.
2. container is running and works well.
![image](https://user-images.githubusercontent.com/7403481/162964523-94a8f829-b48a-476a-bb50-f764fe1e95e7.png)
![image](https://user-images.githubusercontent.com/7403481/162964614-13ea8df0-a0b8-4d7c-bb5d-690aec7b8697.png)

### What did you expect to happen?

1.static pod is running after node reboot

### How can we reproduce it (as minimally and precisely as possible)?

The problem occurs with quite low probability when we performs the following steps:
1. node is in status of memory pressure;
2. node reboot.

### Kubernetes version

<details>
Client version: v1.22.1
Server version: v1.22.1
```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>
NA
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by yangjunmyfm192085** [comment]
Is there any useful information in the kubelet log?

**Comment by ehashman** [comment]
I believe this has been fixed already.

/assign
to add the bug this duplicates, I'll close this when I have that info

**Comment by ehashman** [comment]
I suspect this is https://github.com/kubernetes/kubernetes/issues/105523 which was fixed in 1.22.3.

Please reopen if you upgrade to 1.22.3 and still encounter the issue.

/close
